### PR TITLE
Initial refactor of the documentation

### PR DESCRIPTION
**Pull request by andrea-pasquale**
This PR does the following:
- [x] Remove outdated sections in the doc.
- [x] Sync doc with latest version of Qibolab
- [x] Add all new protocols to the documentation
- [x] Update developer instructions to facilitate the development of new protocols.
- [x] Add scheme with available protocols 

this is not definitive. It is just an initial refactor. More PRs will be opened after this one in order
to have a proper documentation.

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [x] Qibolab_platforms_qrc: `main`

### CODE REVIEW

**Review by Edoardo-Pedicillo — APPROVED** [review]

**Comment by andrea-pasquale** [comment]
@Edoardo-Pedicillo could you please upload here the xmind diagram?

File: doc/source/getting-started/runcard.rst [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
-    * ``runcard``: path to the platform runcard (optional). If not specified it will be used the platform runcard available in qibolab.
-    * ``qubits``: the qubit(s) that we are calibrating.
-    * ``format``: the format for storing the measurements.
-
-After those the user will simply populate ``actions`` with all the routines
-that he will like to run followed by their respective arguments.
+Here is a description of the global parameters to be specified:
+    * ``backend``: ``Qibo`` backend, if not provided ``Qibolab`` will be used.
+    * ``platform``: QPU where the experiments will be executed. Possible choices
+        for TII users are available in this `repository <https:/https://github.com/qiboteam/qibolab_platforms_qrc>`_.
```
Comment (Edoardo-Pedicillo) [inline doc/source/getting-started/runcard.rst:34]:
```suggestion
        for TII users are available in this `repository <https://github.com/qiboteam/qibolab_platforms_qrc>`_.
```

File: doc/source/developer-guides/new_protocol.rst [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+Currently, characterization/calibration protocols in ``Qibocal`` are implemented
+by performing a clear separation between *input parameters*, *data acquired* and
+*results*.
+
+These are then connected through the following functions:
+
+* `acquisition` that receives as input `parameters` and outputs `data`
+* `fit` that receives as input `data` and outputs `results`
+* `plot` that receives as input `data` and `results` to visualize the protocol
+
```
Comment (Edoardo-Pedicillo) [inline doc/source/developer-guides/new_protocol.rst:18]:
```suggestion
Currently, characterization/calibration protocols are divided in three steps: `acquisition`, `fit` and `plot`. ``Qibocal`` provides three data structures  *input parameters*, *data acquired* and
*results*, that collect all the information concerning the routine.

The relationship between steps and data structures are summarized in the following bullets:

* `acquisition` receives as input `parameters` and outputs `data`
* `fit` receives as input `data` and outputs `results`
* `plot` receives as input `data` and `results` to visualize the protocol

```

File: doc/source/developer-guides/new_protocol.rst [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+We believe that this approach is useful in order to code protocols that
+aims at achieving a specific task. However, this approach is flexible enough
+to show some data acquired without performing a post-processing analysis.
```
Comment (Edoardo-Pedicillo) [inline doc/source/developer-guides/new_protocol.rst:21]: The first sentence is not clear, I suggest to drop it.
```suggestion
This approach is flexible enough to allow the data acquisition without performing a post-processing analysis.
```

File: doc/source/developer-guides/new_protocol.rst [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+.. note::
+      It is advised to use `dataclasses`. If you are not familiar
+      have a look at the official `documentation <https://docs.python.org/3/library/dataclasses.html>`_.
+
+
+Data structure
+^^^^^^^^^^^^^^
+Secondly, we define a data structure aims at storing both the angles and
+the probabilities measured for each qubit. A generic data structure is usually composed
+by some raw data (the data attribute), which is usually coded as a dictionary of arrays
```
Comment (Edoardo-Pedicillo) [inline doc/source/developer-guides/new_protocol.rst:68]:
```suggestion
the probabilities measured for each qubit. A generic data structure is usually composed
of some raw data (the data attribute), which is usually coded as a dictionary of arrays
```

File: doc/source/developer-guides/new_protocol.rst [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+.. code-block::python
+
+    from qibolab.qubits import QubitId
+
+    @dataclass
+    class RotationResults(Results):
+        """Results object for data"""
+        fitted_parameters: dict[QubitId, list] = field(default_factory=dict)
+
```
Comment (Edoardo-Pedicillo) [inline doc/source/developer-guides/new_protocol.rst:181]:
```suggestion
.. code-block:: python

    from qibolab.qubits import QubitId

    @dataclass
    class RotationResults(Results):
        """Results object for data"""
        fitted_parameters: dict[QubitId, list] = field(default_factory=dict)

```

**Comment by Edoardo-Pedicillo** [comment]
Thank @andrea-pasquale. If you have time, could you add an image of the report of the `rotation` protocol in the tutorial ?

**Review by scarrazza — APPROVED** [review]

**Pull request merged**